lib/rules/hexadecimal-escape.ts

```diff
@@ -0,0 +1,125 @@
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { Character } from "regexpp/ast"
+import type { RegExpContext } from "../utils"
+import {
+    defineRegexpVisitor,
+    createRule,
+    getEscapeSequenceKind,
+    EscapeSequenceKind,
+    toCharSetSource,
+    isHexadecimalEscape,
+} from "../utils"
+
+export default createRule("hexadecimal-escape", {
+    meta: {
+        docs: {
+            description: "enforce consistent usage of hexadecimal escape",
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [
+            {
+                enum: ["always", "never"], // default always
+            },
+        ],
+        messages: {
+            expectedHexEscape:
+                "Expected hexadecimal escape ('{{hexEscape}}'), but {{unexpectedKind}} escape ('{{rejectEscape}}') is used.",
+            unexpectedHexEscape:
+                "Unexpected hexadecimal escape ('{{hexEscape}}').",
+        },
+        type: "suggestion", // "problem",
+    },
+    create(context) {
+        const always = context.options[0] !== "never"
+
+        /**
+         * Verify for always
+         */
+        function verifyForAlways(
+            { node, getRegexpLocation, fixReplaceNode }: RegExpContext,
+            kind: EscapeSequenceKind,
+            cNode: Character,
+        ) {
+            if (kind === EscapeSequenceKind.hexadecimal) {
+                return
+            }
+
+            const hexEscape = `\\x${cNode.value.toString(16).padStart(2, "0")}`
+
+            context.report({
+                node,
+                loc: getRegexpLocation(cNode),
+                messageId: "expectedHexEscape",
+                data: {
+                    hexEscape,
+                    unexpectedKind: kind,
+                    rejectEscape: cNode.raw,
+                },
+                fix: fixReplaceNode(cNode, hexEscape),
+            })
+        }
+
+        /**
+         * Verify for never
+         */
+        function verifyForNever(
+            { node, getRegexpLocation, fixReplaceNode, flags }: RegExpContext,
+            kind: EscapeSequenceKind,
+            cNode: Character,
+        ) {
+            if (kind !== EscapeSequenceKind.hexadecimal) {
+                return
+            }
+            context.report({
+                node,
+                loc: getRegexpLocation(cNode),
+                messageId: "unexpectedHexEscape",
+                data: {
+                    hexEscape: cNode.raw,
+                },
+                fix: fixReplaceNode(cNode, () => {
+                    let text = toCharSetSource(cNode.value, flags)
+                    if (isHexadecimalEscape(text)) {
```

Comment:
`toCharSetSource` might return a character literal. E.g. for `\x41` this will return just `A`.

Since we know that `cNode` is a hex escape, why not always use `` `\\u00${cNode.raw.slice(2)}` ``?

Reply:
I was testing `\x0a` and found it strange to be converted to` \u000a`. I personally wanted to convert it to `\n`.
But thanks to your comments, I think this is a personal preference and we may need another rule to convert it to `\n`.
I would like to change this rule to use `\\u00${cNode.raw.slice(2)}`

Reply:
I agree that there should be a rule for converting `\x0a` to `\n`. I'll make an issue.